src/pilot/assign_role.py

```diff
@@ -297,52 +297,52 @@ def get_raid_controller_ids(drac_client):
 
 
 def define_controller_logical_disks(drac_client, raid_controller_ids):
-    if len(raid_controller_ids) > 1:
-        LOG.critical(
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
-
-    raid_10_logical_disk = define_single_raid_10_logical_disk(
-        drac_client, raid_controller_ids[0])
+    return _define_controller_or_compute_logical_disks(drac_client,
+                                                       raid_controller_ids)
 
-    # None indicates an error occurred.
-    if raid_10_logical_disk is None:
-        return None
-
-    logical_disks = list()
 
-    # Add the disk to the list only if it is not empty.
-    if raid_10_logical_disk:
-        logical_disks.append(raid_10_logical_disk)
+def define_compute_logical_disks(drac_client, raid_controller_ids):
+    return _define_controller_or_compute_logical_disks(drac_client,
+                                                       raid_controller_ids)
 
-    return logical_disks
 
+def _define_controller_or_compute_logical_disks(drac_client,
+                                                raid_controller_ids):
+    all_physical_disks = drac_client.list_physical_disks()
+    # Get the drives controlled by the RAID controller
+    raid_cntlr_physical_disks=defaultdict(list)
+    for disk in all_physical_disks:
+        if disk.controller in raid_controller_ids:
+            raid_cntlr_physical_disks[disk.controller].append(disk)
 
-def define_compute_logical_disks(drac_client, raid_controller_ids):
-    if len(raid_controller_ids) > 1:
-        LOG.critical(
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+    is_root_volume=True
```

Comment:
let's use `boss_controller` to track if RAID 10 will be root volume or not. This way can avoid introducing another variable. After removing `is_root_volume`, on line 335 use `not boss_controller`.

Reply:
removed _is_root_volume_ and used not _boss_controller_.